### DISCUSSION TITLE
Scraping via contentID?

### DISCUSSION THREAD

**Discussion by endofthescope** [javinizer/Javinizer#289] [Q&A]
Hello, is there a good way to scrape via contentID instead of DVD ID? A lot of the JAV I download comes in through content ID, and I always used that to scrape previously because it gave me better results on JAVmoviescraper. I tried scraping with contentID on JAVinizer but wasn't getting any matches. Is there a setting I can change, or if not maybe it could be a feature I can request? Thank you for your effort and time. 

and so it's clear, content ID is listed on R18 and I think DMM, the content ID for JUFE-287 would be jufe00287. It's also what's used in the URL of the video page.

**Comment by jvlflame** [comment]
There's currently some basic conversion between contentId and dvdId values being done by the file matcher. You can see this output if you run the command `Javinizer -Path path/to/jav/files -Preview`. Unfortunately I don't think I have all cases matching properly, such as if the contentId has numbers or other alphanumeric characters prepending the actual id.

If you can provide me a few examples of contentId values which are currently not being matched properly I'll take a look into fixing it.

**Reply by endofthescope** [reply]
here is the readout of that command on 2 files:

```
Id         : 1DANDAN-004
ContentId  : 1DANDAN00004
FileName   : 1dandan00004.mkv
BaseName   : 1dandan00004
Directory  : F:\dl\jj\1\2
FullName   : F:\dl\jj\1\2\1dandan00004.mkv
Extension  : .mkv
Length     : 4617.46
PartNumber :

Id         : XVSR-606
ContentId  : XVSR00606
FileName   : xvsr00606.mkv
BaseName   : xvsr00606
Directory  : F:\dl\jj\1\2
FullName   : F:\dl\jj\1\2\xvsr00606.mkv
Extension  : .mkv
Length     : 4967.75
PartNumber :
```


I tried these 4 filenames:

1dandan00004.mkv
1sdnm00298.mkv
vec00487.mkv
xvsr00606.mkv

and actually the last 2 were scraping the information correctly, so another good start, but the top 2 didn't find a match. To test if it's because of the prefix, I scraped this file, which also has a prefix: 

h_1100hzgd00186.mkv - hzgd-186

and no match. So it seems like that is causing it. I think I thought "none" were working because I was starting at the top of my list, and the ones with number prefixes are at the top of the list so I wasn't getting results. You can also see in the powershell readout that it's getting "1DANDAN-004", which brings no results on R18 (site that I'm scraping on).

and for more info in case it's useful I'm using the GUI on windows 10 and I'm scraping R18 exclusively. Thank you!

edit: I changed this post because I didn't fully read your comment at first, woops!

**Reply by jvlflame** [reply]
Thanks for the details.

I added a regex check to better match contentid for the next release. Hopefully this should resolve the majority of failing matches, but let me know if you find any additional filenames that fail to match.
`([a-zA-Z]?_?\d+)?([a-zA-Z]+\d+)`

![image](https://user-images.githubusercontent.com/17932235/130371425-598c1724-0726-4d0e-93db-ec59c7599180.png)
